example/javalib/android/1-hello-world/build.mill

```diff
@@ -0,0 +1,137 @@
+import mill._
+import mill.scalalib._
+import mill.api.PathRef
+import requests._
+import mill.javalib.android.AndroidModule
+
+object app extends AndroidModule {
+
+  val currentFolder = os.Path(millSourcePath.toString.replace("app", ""))
+  val appName = "HelloWorld"
+  
+  def generateResources = T {
+    installAndroidSdk()
+    val genDir = T.dest / "gen"
+    os.makeDir.all(genDir)
+    os.proc(
+      aapt,
+      "package",
+      "-f",
+      "-m",
+      "-J", genDir,
+      "-M", currentFolder / "AndroidManifest.xml",
+      "-I", androidJarPath
+    ).call()
+    PathRef(genDir)
+  }
+
+  def compileJava = T {
+    val objDir = T.dest / "obj"
+    os.makeDir.all(objDir)
+    os.proc(
+      "javac",
+      "-classpath", androidJarPath,
+      "-d", objDir,
+      os.walk(generateResources().path).filter(_.ext == "java"),
+      os.walk(currentFolder / "java").filter(_.ext == "java")
+    ).call()
+    PathRef(objDir)
+  }
+
+  def createJar = T {
+    val jarPath = T.dest / "my_classes.jar"
+    os.proc(
+      d8,
+      os.walk(compileJava().path).filter(_.ext == "class"),  
+      "--output", jarPath,
+      "--no-desugaring"  
+    ).call()
+    PathRef(jarPath)
+  }
+
+  def createDex = T {
+    val dexPath = T.dest
+    os.proc(
+      d8,
+      createJar().path,
+      androidJarPath,
+      "--output", dexPath
+    ).call()
+    PathRef(dexPath)
+  }
+
+  def createApk = T {
+    val apkPath = T.dest / s"${appName}.unsigned.apk"
+    os.proc(
+      aapt,
+      "package",
+      "-f",
+      "-M", currentFolder / "AndroidManifest.xml",
+      "-I", androidJarPath,
+      "-F", apkPath,
```

Comment:
Are there long flags we can use here?

Reply:
yes there are available long flags also do we need to use ?

Reply:
yes